plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveSplitSource.java

```diff
@@ -305,7 +311,7 @@ else if (maxSplitBytes * 2 >= remainingBlockBytes) {
                         internalSplit.getFileModifiedTime(),
                         internalSplit.getSchema(),
                         internalSplit.getPartitionKeys(),
-                        block.getAddresses(),
+                        cachingHostAddressProvider.getHosts(internalSplit.getPath(), block.getAddresses()),
```

Comment:
Block addresses are populated when the filesystem is HDFS. When caching is used with HDFS, we still want caching to drive split scheduling decision rather than HDFS block locality.